charts/platform-tenant/templates/components/apps/_virtual-service.tpl

```diff
@@ -21,6 +21,9 @@ http:
 {{- define "platform-tenant.app.virtual-service.template" }}
 {{- $ := last . }}
 {{- $app := first . }}
+{{- $istio := $.Values.components.istio -}}
+{{- $gwNamespace := default (printf "%s-gateway" $istio.namespace) $istio.components.gateway.namespace -}}
```

Comment:
`$gatewayNamespace`, no need for brevity. Don't default to `$istio.namespace` as gateways should never be in the same namespace as istiod.